silkrpc/commands/trace_api.cpp

```diff
@@ -130,12 +130,41 @@ asio::awaitable<void> TraceRpcApi::handle_trace_raw_transaction(const nlohmann::
 
 // https://eth.wiki/json-rpc/API#trace_replayblocktransactions
 asio::awaitable<void> TraceRpcApi::handle_trace_replay_block_transactions(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() < 2) {
+        auto error_msg = "invalid trace_replayBlockTransactions params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    const auto block_number_or_hash = params[0].get<BlockNumberOrHash>();
+    const auto trace_type = params[1].get<std::vector<std::string>>();
+
+    trace::TraceConfig config;
+    for (auto entry : trace_type) {
+        if (entry == "trace") {
+            config.trace = true;
+        }
+        if (entry == "vmTrace") {
+            config.vm_trace = true;
+        }
+        if (entry == "stateDiff") {
+            config.state_diff = true;
+        }
+    }
```

Comment:
I think that adding the definition of `from_json` function for the type `trace::TraceConfig` would allow to avoid this code repetition in all `trace_` API functions.

After such change the code should become: `const auto config = params[1].get<trace::TraceConfig>();`

Reply:
Great suggestion!